### PR TITLE
Allow constant nodes (not only constant string expressions) to be used in IsMatch

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
This was the previous behavior - we should revert to what we had before. Since IsMatch doesn't need to know the value of the regular expressions at compile-time (unlike Match[All], which need it to determine the output schema), the constant check can be less strict.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using Microsoft.PowerFx.Core.Types;
 using Microsoft.PowerFx.Core.Utils;
 using Microsoft.PowerFx.Syntax;
+using static Microsoft.PowerFx.Syntax.PrettyPrintVisitor;
```
Comment (nguhoa) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs:13]: i think you don't need this.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs [hunk 2]
```diff
@@ -41,19 +42,12 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
...
 
-        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
         {
-            bool fValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);            
-
-            TexlNode regExNode = args[1];
-            
-            if ((argTypes[1].Kind != DKind.String && argTypes[1].Kind != DKind.OptionSetValue) || !BinderUtils.TryGetConstantValue(context, regExNode, out var nodeValue))                
+            if ((argTypes[1].Kind != DKind.String && argTypes[1].Kind != DKind.OptionSetValue) || !binding.IsConstant(args[1]))
```
Comment (nguhoa) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs:45]: >argTypes[1] [](http://example.com/codeflow?start=17&length=11)

are we sure that argTypes is not null and length is more than 1? should we check before refer index 1?

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs:45]: The function arity (defined in the constructor) defines that it should have at least 2 (and at most 3) arguments, so argTypes.Length is guaranteed to be between 2 and 3.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**